scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,327 @@
+                promise.resolve(value);
+            }
+        });
+        response.on("error", function (e) {
+            fluid.log(errorMsg + e.message);
+            promise.reject(e);
+        });
+    };
+};
+
+/**
+ * Quit the whole process because a request of the database has failed, and log
+ * the error.  Use this function when the failure has not actually modified the
+ * database; for example, when getting all the current snapset Prefs Safes.  If
+ * that failed, that database is as it was, but there is no point in continuing.
+ * @param {String} errorMsg - The reason why database access failed.
+ */
+dbLoader.bail = function (errorMsg) {
+    fluid.log (errorMsg);
+    process.exit(1);
+};
+
+/**
+ * General mechanism to create a database request, set up an error handler and
+ * return.  It is up to the caller to trigger the request by calling its end()
+ * function.
+ * @param {String} databaseURL - URL to query the database with.
+ * @param {Function} handleResponse - callback that processes the response from
+ *                                    the request.
+ * @param {String} errorMsg - optional error message for request errors.
+ * @return {http.ClientRequest} - The http request object.
+ */
+dbLoader.queryDatabase = function (databaseURL, handleResponse, errorMsg) {
+    var aRequest = http.request(databaseURL, handleResponse);
+    aRequest.on("error", function (e) {
+        fluid.log(errorMsg + e.message);
+    });
+    return aRequest;
+};
+
+/**
+ * Get all the json files from the given directory, then loop to put their
+ * contents into an array of Objects.
+ * @param {String} dataDir - Directory containing the files to load.
+ * @return {Array} - Each element of the array is an Object based on the
+ *                   contents of each file loaded.
+ */
+dbLoader.getDataFromDirectory = function (dataDir) {
+    var contentArray = [];
+    var files = fs.readdirSync(dataDir);
+    files.forEach(function (aFile) {
+        if (aFile.endsWith(".json")) {
+            var fileContent = fs.readFileSync(dataDir + "/" + aFile, "utf-8");
+            contentArray = contentArray.concat(JSON.parse(fileContent));
+        }
+    });
+    return contentArray;
+};
+
+// Load the static data first.  If the database is fresh, there won't be any
+// views to use to find the snapset Prefs Safes.  This ensures they are loaded
+// regardless.  If the data base already has these views, then this is
+// essentially a no-op.
+var staticData = dbLoader.getDataFromDirectory(dbLoader.staticDataDir);
+var staticDataPromise = fluid.promise();
+var staticDataResponse = dbLoader.createResponseHandler(
+    function () {
+        fluid.log("Loading static data from '" + dbLoader.staticDataDir + "'");
+    },
+    staticDataPromise
+);
+var execStaticDataRequest = dbLoader.createBulkDocsRequest(staticData, staticDataResponse);
+execStaticDataRequest();
+
+// Secondly, get the snapsets Prefs Safes.
+var snapsetsPromise = fluid.promise();
```

Comment:
Please write functions which produce these rather than assigning them as globals. Then write a separate function which orchestrates them together in a sequence.

Reply:
Done.